packages/infra/lib/ihe-stack.ts

```diff
@@ -0,0 +1,92 @@
+import { CfnOutput, StackProps } from "aws-cdk-lib";
+import * as apig from "aws-cdk-lib/aws-apigateway";
+import * as cert from "aws-cdk-lib/aws-certificatemanager";
+import { SnsAction } from "aws-cdk-lib/aws-cloudwatch-actions";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as r53 from "aws-cdk-lib/aws-route53";
+import * as r53_targets from "aws-cdk-lib/aws-route53-targets";
+import { Construct } from "constructs";
+import { EnvConfig } from "../config/env-config";
+import { createLambda } from "./shared/lambda";
+
+interface IHEStackProps extends StackProps {
+  config: EnvConfig;
+  vpc: ec2.IVpc;
+  alarmAction: SnsAction | undefined;
+  lambdaLayers: lambda.ILayerVersion[];
+  certificate: cert.DnsValidatedCertificate;
+  publicZone: r53.IHostedZone;
+}
+
+export function createIHEStack(stack: Construct, props: IHEStackProps) {
+  //-------------------------------------------
+  // API Gateway
+  //-------------------------------------------
+  if (!props.config.iheSubdomain) {
+    throw new Error("Must define iheSubdomain if building the IHE stack!");
```

Comment:
Makes sense, but this would make it easier to slip under the radar

ie the caller of the `create` function could be confused as to why it's not deploying anything when it's explicitly called - if it's cool, I'd keep this one as-is